src/auth/webhook-secret.guard.ts

```diff
@@ -0,0 +1,32 @@
+import {
+  CanActivate,
+  ExecutionContext,
+  Injectable,
+  UnauthorizedException,
+} from '@nestjs/common';
+import * as crypto from 'crypto';
+import { Request } from 'express';
+import { Env } from 'src/env';
+
+@Injectable()
+export class WebhookSecretGuard implements CanActivate {
+  canActivate(context: ExecutionContext): boolean {
+    const request: Request = context.switchToHttp().getRequest();
+    const signature = request.headers['x-hub-signature-256'] as string;
+    const secret = Env.SECRET.WEBHOOK;
```

Comment:
PullProofAI

There is a potential security vulnerability in the `WebhookSecretGuard` class. Storing sensitive information like secrets directly in the code can lead to exposure if the code repository is compromised.

It's recommended to avoid hardcoding secrets in the source code. Instead, consider using environment variables or a secure configuration management solution to store and retrieve sensitive information.

```diff
-     const secret = Env.SECRET.WEBHOOK;
+     const secret = process.env.WEBHOOK_SECRET;
```

#### End of Review


<!-- This is an auto-generated comment by PullProofAI -->